pkg/plugins/plugins.go

```diff
@@ -0,0 +1,168 @@
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "invalid method: only GET is allowed"})
+		return
+	}
+
+	pluginName, pluginFilePath := parsePluginNameAndPath(r.URL.Path)
+	pluginRequest, err := p.getServiceRequestURL(pluginName, pluginFilePath)
+	if err != nil {
+		errMsg := err.Error()
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	req, err := http.NewRequest("GET", pluginRequest, nil)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to create GET request for %q plugin: %v", pluginName, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	resp, err := p.Client.Do(req)
+	if err != nil {
+		errMsg := fmt.Sprintf("GET request for %q plugin failed: %v", pluginName, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		err := fmt.Errorf("console service account cannot list resource: %s", resp.Status)
+		serverutils.SendResponse(w, resp.StatusCode, serverutils.ApiError{Err: err.Error()})
+		return
+	}
+
+	// copy 'Content-Type' header from the plugin's server response
+	responseContentType := resp.Header.Get("Content-Type")
+	if responseContentType != "" {
+		w.Header().Set("Content-Type", responseContentType)
+	}
+
+	w.WriteHeader(resp.StatusCode)
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to read HTTP response body from %q plugin: %v", pluginName, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	_, err = w.Write(bodyBytes)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed sending HTTP response body from %q plugin: %v", pluginName, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+}
+
+func (p *PluginsHandler) SetPluginsEndpointMap() error {
+	pluginsList := ConsolePluginList{}
+	plugins, err := p.fetchPlugins()
+	if err != nil {
+		klog.Errorln(err)
+		return err
+	}
+	if err := json.Unmarshal(plugins, &pluginsList); err != nil {
+		return err
+	}
+	pluginsEndpointMap := map[string]string{}
+	for _, plugin := range pluginsList.Items {
+		pluginsEndpointMap[plugin.Name] = getServiceHostname(plugin)
+	}
+	p.PluginsEndpointMap = pluginsEndpointMap
+
+	return nil
+}
+
+func (p *PluginsHandler) getServiceRequestURL(pluginName, pluginFilePath string) (string, error) {
+	if !contains(p.EnabledPlugins, pluginName) {
+		return "", fmt.Errorf("bad request: plugin %q is not enabled", pluginName)
+	}
+
+	pluginEndpoint, ok := p.PluginsEndpointMap[pluginName]
+	if ok != true {
+		return "", fmt.Errorf("failed to create GET request: could not get endpoint for %q plugin", pluginName)
+	}
```

Comment:
do you really need `EnabledPlugins`? Wouldn't `pluginEndpoint, ok := p.PluginsEndpointMap[pluginName]; ok` be enough?

Reply:
yes, cause if no ConsolePlugin is enabled then we dont need to register the handler for `/api/plugins/` endpoint.